lib/mix/test/mix/cli_test.exs

```diff
@@ -43,13 +43,16 @@ defmodule Mix.CLITest do
 
         def run(_) do
           IO.puts Mix.Project.get!.hello_world
+          Mix.shell.info("This won't appear")
         end
       end
       """
 
-      contents = mix ~w[my_hello]
+      contents = mix ~w[my_hello], [{"MIX_QUIET", "1"}]
+
       assert contents =~ "Hello from MyProject!\n"
-      assert contents =~ "Compiled lib/hello.ex\n"
+      refute contents =~ "Compiled lib/hello.ex\n"
```

Comment:
No it doesn't. We can even get rid of the line, honestly. :) we know it was
compiled.

On Sunday, October 25, 2015, Maurizio Del Corno <notifications@github.com>
wrote:

> In lib/mix/test/mix/cli_test.exs
> <https://github.com/elixir-lang/elixir/pull/3910#discussion_r42942371>:
>
> >        assert contents =~ "Hello from MyProject!\n"
> > -      assert contents =~ "Compiled lib/hello.ex\n"
> > +      refute contents =~ "Compiled lib/hello.ex\n"
>
> Ok @josevalim <https://github.com/josevalim>, changes done. My only doubt
> is that I had to change this line from assert to refute or this test
> won't pass (the env var changes the Mix.shell, so "Compiled lib/hello.ex"
> doesn't appear). I think that this doesn't change the validity of the test
> (compilation is tested in the previous tests), but I'll wait for your
> opinion. [image: :+1:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/3910/files#r42942371>.
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*